### ISSUE TITLE
Fresh install 2.5.0 failed

### ISSUE THREAD

**Issue by goldsky** [modxcms/revolution#12987]
### Summary
Fresh installation for 2.5.0 failed because `modx_system_settings` table was not created successfully.

### Step to reproduce
Fresh install

### Observed behavior
It went to 
```
500 Error

Site temporarily unavailable.
```
page.
The installation log:

```
[2016-04-27 17:30:46] (ERROR @ /var/www/html/***/www/core/xpdo/om/xpdoobject.class.php : 240) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 17:30:46] (ERROR @ /var/www/html/***/www/core/xpdo/om/xpdoobject.class.php : 240) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 17:30:46] (FATAL) Could not load core MODX configuration!
```

### Expected behavior
Should continue to summary page.

### Environment
MODX version 2.5.0,
Apache 2.0,
PHP Version 5.6.20-1+deb.sury.org~wily+1
mysqlnd 5.0.11-dev - 20120503

**Comment by goldsky** [comment]
Looks like the problem exists since the previous versions, because I've tested way back until 2.3.x
The install log:
```
[2016-04-27 18:04:10] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_context_setting`
SQL: CREATE TABLE `modx_context_setting` (`context_key` VARCHAR(255) NOT NULL, `key` VARCHAR(50) NOT NULL, `value` MEDIUMTEXT NULL, `xtype` VARCHAR(75) NOT NULL DEFAULT 'textfield', `namespace` VARCHAR(40) NOT NULL DEFAULT 'core', `area` VARCHAR(255) NOT NULL DEFAULT '', `editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP, PRIMARY KEY (`context_key`,`key`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'editedon'
)

[2016-04-27 18:04:11] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_lexicon_entries`
SQL: CREATE TABLE `modx_lexicon_entries` (`id` INTEGER unsigned NOT NULL AUTO_INCREMENT, `name` VARCHAR(255) NOT NULL DEFAULT '', `value` TEXT NOT NULL, `topic` VARCHAR(255) NOT NULL DEFAULT 'default', `namespace` VARCHAR(40) NOT NULL DEFAULT 'core', `language` VARCHAR(20) NOT NULL DEFAULT 'en', `createdon` DATETIME NULL, `editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP, PRIMARY KEY (`id`), INDEX `name` (`name`), INDEX `topic` (`topic`), INDEX `namespace` (`namespace`), INDEX `language` (`language`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'editedon'
)

[2016-04-27 18:04:11] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_manager_log`
SQL: CREATE TABLE `modx_manager_log` (`id` INTEGER unsigned NOT NULL AUTO_INCREMENT, `user` INT(10) unsigned NOT NULL DEFAULT '0', `occurred` DATETIME NULL DEFAULT '0000-00-00 00:00:00', `action` VARCHAR(100) NOT NULL DEFAULT '', `classKey` VARCHAR(100) NOT NULL DEFAULT '', `item` VARCHAR(255) NOT NULL DEFAULT '0', PRIMARY KEY (`id`), INDEX `user_occurred` (`user`,`occurred`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'occurred'
)

[2016-04-27 18:04:11] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_system_settings`
SQL: CREATE TABLE `modx_system_settings` (`key` VARCHAR(50) NOT NULL DEFAULT '', `value` TEXT NOT NULL, `xtype` VARCHAR(75) NOT NULL DEFAULT 'textfield', `namespace` VARCHAR(40) NOT NULL DEFAULT 'core', `area` VARCHAR(255) NOT NULL DEFAULT '', `editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP, PRIMARY KEY (`key`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'editedon'
)

[2016-04-27 18:04:12] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_user_group_settings`
SQL: CREATE TABLE `modx_user_group_settings` (`group` INTEGER unsigned NOT NULL DEFAULT '0', `key` VARCHAR(50) NOT NULL, `value` TEXT NULL, `xtype` VARCHAR(75) NOT NULL DEFAULT 'textfield', `namespace` VARCHAR(40) NOT NULL DEFAULT 'core', `area` VARCHAR(255) NOT NULL DEFAULT '', `editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP, PRIMARY KEY (`group`,`key`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'editedon'
)

[2016-04-27 18:04:12] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_user_messages`
SQL: CREATE TABLE `modx_user_messages` (`id` INTEGER unsigned NOT NULL AUTO_INCREMENT, `type` VARCHAR(15) NOT NULL DEFAULT '', `subject` VARCHAR(255) NOT NULL DEFAULT '', `message` TEXT NOT NULL, `sender` INT(10) NOT NULL DEFAULT '0', `recipient` INT(10) NOT NULL DEFAULT '0', `private` TINYINT(4) NOT NULL DEFAULT '0', `date_sent` DATETIME NOT NULL DEFAULT '0000-00-00 00:00:00', `read` TINYINT(1) NOT NULL DEFAULT '0', PRIMARY KEY (`id`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'date_sent'
)

[2016-04-27 18:04:12] (ERROR @ /var/www/html/***/www/setup/includes/tables_create.php : 116) Could not create table `modx_user_settings`
SQL: CREATE TABLE `modx_user_settings` (`user` INT(11) NOT NULL DEFAULT '0', `key` VARCHAR(50) NOT NULL DEFAULT '', `value` TEXT NULL, `xtype` VARCHAR(75) NOT NULL DEFAULT 'textfield', `namespace` VARCHAR(40) NOT NULL DEFAULT 'core', `area` VARCHAR(255) NOT NULL DEFAULT '', `editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP, PRIMARY KEY (`user`,`key`)) ENGINE=MyISAM
ERROR: Array
(
    [0] => 42000
    [1] => 1067
    [2] => Invalid default value for 'editedon'
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/om/xpdoobject.class.php : 424) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/transport/xpdoobjectvehicle.class.php : 202) Error 42S02 executing statement:
INSERT INTO `modx_system_settings` (`key`, `value`, `xtype`, `namespace`, `area`) VALUES ('access_category_enabled', '1', 'combo-boolean', 'core', 'authentication')
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/transport/xpdoobjectvehicle.class.php : 88) Error saving vehicle object of class modSystemSetting; criteria: Array
(
    [key] => access_category_enabled
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/om/xpdoobject.class.php : 424) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/transport/xpdoobjectvehicle.class.php : 202) Error 42S02 executing statement:
INSERT INTO `modx_system_settings` (`key`, `value`, `xtype`, `namespace`, `area`) VALUES ('access_context_enabled', '1', 'combo-boolean', 'core', 'authentication')
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/transport/xpdoobjectvehicle.class.php : 88) Error saving vehicle object of class modSystemSetting; criteria: Array
(
    [key] => access_context_enabled
)

[2016-04-27 18:04:13] (ERROR @ /var/www/html/***/www/core/xpdo/om/xpdoobject.class.php : 424) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'modx_***.modx_system_settings' doesn't exist
)
```
.... error continues for inserting System settings to non existed "modx_system_settings" table

**Comment by OptimusCrime** [comment]
Strict mode turned on?

**Comment by Jako** [comment]
Duplicate of #12838

**Comment by goldsky** [comment]
@OptimusCrime 
Yep.
```
mysql> select @@GLOBAL.sql_mode;
+-------------------------------------------------------------------------------------------------------------------------------------------+
| @@GLOBAL.sql_mode                                                                                                                         |
+-------------------------------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+-------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0,00 sec)
```

**Comment by goldsky** [comment]
Hmmm but the problem persists although I've removed the strict mode.

```
mysql> set @@global.sql_mode='';
Query OK, 0 rows affected, 1 warning (0,00 sec)

mysql> select @@GLOBAL.sql_mode;
+-------------------+
| @@GLOBAL.sql_mode |
+-------------------+
|                   |
+-------------------+
1 row in set (0,00 sec)

mysql> exit
Bye
me@work:/etc/mysql$ su
Password: 
root@work:/etc/mysql# service mysql restart

```

**Comment by goldsky** [comment]
I found out that the installation worked fine for MySQL 5.5.33.
Mine is 5.7.12

**Comment by Bruno17** [comment]
`editedon` TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP
yeah, seems editedon is defined with NOT NULL but doesn't have a definition for default ?

**Comment by sottwell** [comment]
There are quite a few of those in the setup schemas.

**Comment by Bruno17** [comment]
A PR with fixed schemas, if that's the case, anyone?

**Comment by sottwell** [comment]
And perhaps along with it, fix the issue with the length of text fields used for keys for utf8mb4 charsets https://github.com/modxcms/revolution/issues/12445

**Comment by Jako** [comment]
@goldsky Maybe this helps: https://mattstauffer.co/blog/how-to-disable-mysql-strict-mode-on-laravel-forge-ubuntu

Removing STRICT_TRANS_TABLES should it be at the moment from sql_mode.

But fixing those incomplete schemas would be great.

**Comment by sottwell** [comment]
Did a pull request, needs testing by somebody with STRICT_TRANS_TABLES enabled.

**Comment by sottwell** [comment]
The failed defaults for the datetime fields are caused by a mismatch between what PHP thinks the format of a datetime should be and what MySQL wants. This can depend on server settings, locale settings and anything else that affects the formatting of a datetime.

"This is caused by Zend not setting your timestamp format to one that matches what MySQL is expecting. You could disable STRICT mode in MySQL, but this is a hack and not a solution (MySQL will attempt to guess what the date you're entering is)."

**Comment by goldsky** [comment]
@Jako nope, even leaving it blank doesn't work.
MySQL 5.7 brings specific changes on [SQL Mode](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sql-mode-changes).

**Comment by OptimusCrime** [comment]
@goldsky Try the pull request from @sottwell. It looks like it could fix this problem.

**Comment by pixelchutes** [comment]
@goldsky Confirmed this is now fixed in HEAD of `2.5.x`. Try now!

**Comment by goldsky** [comment]
after I pulled the repo, and replaced core/model/ entirely, it worked.
thanks!

**Comment by opengeek** [comment]
@pixelchutes if this is solved in HEAD, why is this still a problem in 2.5.1?

**Comment by pixelchutes** [comment]
@opengeek Where are you seeing this problem exactly? Issue #?

**Comment by opengeek** [comment]
I was seeing it on a fresh install of 2.5.1 with MariaDB 10.1.17 — removing all the strict modes seems to have resolved it and allowed me to install all tables successfully. I'm little confused on whether disabling strict modes was supposed to be necessary I suppose.

**Comment by pixelchutes** [comment]
Interesting. I've not tested on MariaDB, perhaps something about [the fix](5324f9245a59c07abfa6f39c1d05673483cd89e5) does not cover it? I know MySQL 5.6 [w/ strict mode was tested](https://github.com/modxcms/revolution/pull/13001#issuecomment-217294100), as well 5.7 with default strict mode settings, too.
